core/src/HIP/Kokkos_HIP_Space.hpp

```diff
@@ -375,7 +375,28 @@ struct MemorySpaceAccess<HIPManagedSpace, HIPHostPinnedSpace> {
   enum : bool { deepcopy = true };
 };
 
+template <typename MemorySpace>
+void runtime_check_memory_space(const void* ptr, const MemorySpace&);
```

Comment:
Since we check for assignability, should it be represented in the name?

Reply:
Sure, see 0a911e4e49313f1005145c7682292fb4d6b7cad9.